elements/lisk-tree/src/sparse_merkle_tree/utils.ts

```diff
@@ -13,9 +13,180 @@
  */
 
 import { hash } from '@liskhq/lisk-cryptography';
-import { LEAF_HASH_PREFIX, BRANCH_HASH_PREFIX } from './constants';
+import { objects } from '@liskhq/lisk-utils';
+import { BRANCH_HASH_PREFIX, EMPTY_HASH, LEAF_HASH_PREFIX } from './constants';
+import { InclusionProofQuery, InclusionProofQueryWithHash } from './types';
+
+// Sort queries by the longest binaryBitmap, breaking ties by smaller key.
+// https://github.com/LiskHQ/lips-staging/blob/master/proposals/lip-0039.md#proof-construction
+export const sortByBitmapAndKey = <T extends InclusionProofQuery>(queries: T[]): T[] =>
+	queries.sort((q1, q2) => {
+		const q1BinaryBitmap = bufferToBinaryString(q1.bitmap);
+		const q2BinaryBitmap = bufferToBinaryString(q2.bitmap);
+
+		if (q1BinaryBitmap.length === q2BinaryBitmap.length) {
+			return q1.key.byteLength - q2.key.byteLength;
+		}
+
+		return q2BinaryBitmap.length - q1BinaryBitmap.length;
```

Comment:
maybe isolate the logic here as a separate function for better readability

Reply:
Didn't understood. Which logic to move to separate function?

Reply:
I was thinking to isolate this code block but it's not that important.
```
		if (q1BinaryBitmap.length === q2BinaryBitmap.length) {
			return q1.key.byteLength - q2.key.byteLength;
		}

		return q2BinaryBitmap.length - q1BinaryBitmap.length;
```